kotlin-inject-compiler/ksp/src/main/kotlin/me/tatarka/inject/compiler/ksp/ProcessKmpComponentCreate.kt

```diff
@@ -27,10 +26,22 @@ internal fun processKmpComponentCreate(
     val returnTypeClass = returnType.resolvedType().toAstClass()
     if (!astFunction.validateReturnType(returnTypeClass, provider)) return true
 
-    process(astFunction, returnTypeClass, codeGenerator, kmpComponentCreateGenerator)
+    kmpComponentCreateFunctionsByComponentType.getOrPut(returnTypeClass, ::ArrayList).add(astFunction)
+
     true
 }
 
+internal fun generateKmpComponentCreateFiles(
+    codeGenerator: CodeGenerator,
+    generator: KmpComponentCreateGenerator,
+    kmpComponentCreateFunctionsByComponentType: Map<AstClass, List<AstFunction>>
+) {
+    kmpComponentCreateFunctionsByComponentType.forEach { (componentType, kmpComponentCreateFunctions) ->
+        val file = generator.generate(componentType, kmpComponentCreateFunctions)
+        file.writeTo(codeGenerator, aggregating = false)
```

Comment:
This would need to be aggregating now right? Since there could be multiple create functions for the same component in different source files. My suggestion to group by file would allow it to be kept as non-aggregating, though I'm not sure there's a meaningful different in performance.

Reply:
Good catch, yes this would need to be aggregating. I could switch to grouping by file, but I think that loses some of the expressiveness of the file name. In this case I think aggregation would be ammortized away since the inject files are aggregating anyways, and they'd likely be closely related. What do you think?

Reply:
>  think that loses some of the expressiveness of the file name
I'm not sure what you mean by that?
> since the inject files are aggregating anyways
component generation is not aggregating, since there's a file generated for each component, unless you meant something else?

Reply:
> > think that loses some of the expressiveness of the file name

> I'm not sure what you mean by that?

For example, I have a file called `SearchPortal.kt` that contains a `@Component abstract class SearchComponent` and `@KmpComponentCreate expect fun SearchComponent.Companion.createKmp`.

The generated file for the `Component` is `InjectSearchComponent.kt` which lets me know that this is the generated file related to `SearchComponent`.

With my current changes, the generated file for `KmpComponentCreate` would be `KmpComponentCreateSearchComponent.kt` which lets me know that this is the generated file related to `SearchComponent`.

If we switch it to be source file based the generated file would be `KmpComponentCreateSearchPortal.kt` which doesn't necessarily indicate that it is related to `SearchComponent` (the source file might be entirely unrelated, there might be multiple `Components` in the file, etc...).

> > since the inject files are aggregating anyways

> component generation is not aggregating, since there's a file generated for each component, unless you meant something else?

Doesn't [this](https://github.com/evant/kotlin-inject/blob/main/kotlin-inject-compiler/ksp/src/main/kotlin/me/tatarka/inject/compiler/ksp/ProcessInject.kt#L79) indicate that it is aggregating, or am I confusing something?

Reply:
1. Yeah I guess that's true, I don't put that much weight in them since it's not that all that likely you'd be looking at the generated filenames directly but not against making them better.
2. 🤦‍♀️ should've looked before saying that, I don't think it was originally but there was probably a reason to changing to that. "aggregation would be ammortized away" is probably true then.

Reply:
OK, so is this good to merge? I pushed a change setting `aggregating = true`.

Reply:
yeah